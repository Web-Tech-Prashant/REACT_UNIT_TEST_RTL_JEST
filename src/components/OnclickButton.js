import { useState } from "react"

export const ButtonComponent = ()=>{
    const[name,setName] = useState("");

    return(
        <div>
        <button type="button" onClick={()=>setName("Prashant Kumar Singh")}>Display My Name</button>
        <p>{name ? `Hi ${name}`: ""}</p>
        </div>
    )
}