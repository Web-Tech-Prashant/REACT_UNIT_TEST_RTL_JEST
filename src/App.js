import logo from "./logo.svg";
import "./App.css";
import { Sum } from "./components/Sum";
import InputComponent from "./components/OnchangeInput";
import { ButtonComponent } from "./components/OnclickButton";

function App() {
  return (
    // <div className="App">
    //   <h4>Total Sum= {Sum(2, 3)}</h4>
    //   <input
    //     type="text"
    //     placeholder="Enter UserName"
    //     name="username"
    //     id="userId"
    //     value="Prashant"
    //     readOnly
    //   />
    //   <img
    //     title="FirstImage"
    //     alt="photographer"
    //     src="https://buffer.com/cdn-cgi/image/w=1000,fit=contain,q=90,f=auto/library/content/images/size/w600/2023/10/free-images.jpg"
    //   ></img>
    // </div>
    <div style={{margin:"20px 35px 35px 35px"}}>
      <InputComponent/>{" "}
      <ButtonComponent/>
    </div>
  );
}

export default App;
