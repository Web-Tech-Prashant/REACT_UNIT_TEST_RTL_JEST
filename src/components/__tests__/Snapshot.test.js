


import { render,screen } from "@testing-library/react";
import {expect,test,describe} from "@jest/globals";
import InputComponent from "../OnchangeInput";
import { ButtonComponent } from "../OnclickButton";
import App from "../../App";

describe("Snapshot Testing Group",()=>{
    test("Snapshot-1",()=>{
       const container = render(<InputComponent/>);
       expect(container).toMatchSnapshot();
    })
    test("Snapshot-2",()=>{
        const container1 = render(<ButtonComponent/>);
        expect(container1).toMatchSnapshot();
    })
    // test("Snapshot-3",()=>{
    //     const container2 = render(<App/>);
    //     expect(container2).toMatchSnapshot();
    // })
})