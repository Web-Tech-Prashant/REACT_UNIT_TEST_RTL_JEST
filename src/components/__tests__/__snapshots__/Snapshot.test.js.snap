// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Group Snapshot-1 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label>
        <b>
          UserName:
        </b>
         
        <input
          id="userId"
          name="name"
          placeholder="Enter User Name"
          type="text"
          value=""
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label>
      <b>
        UserName:
      </b>
       
      <input
        id="userId"
        name="name"
        placeholder="Enter User Name"
        type="text"
        value=""
      />
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Testing Group Snapshot-2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          type="button"
        >
          Display My Name
        </button>
        <p />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        type="button"
      >
        Display My Name
      </button>
      <p />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Snapshot Testing Group Snapshot-3 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="margin: 20px 35px 35px 35px;"
      >
        <label>
          <b>
            UserName:
          </b>
           
          <input
            id="userId"
            name="name"
            placeholder="Enter User Name"
            type="text"
            value=""
          />
        </label>
         
        <div>
          <button
            type="button"
          >
            Display My Name
          </button>
          <p />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="margin: 20px 35px 35px 35px;"
    >
      <label>
        <b>
          UserName:
        </b>
         
        <input
          id="userId"
          name="name"
          placeholder="Enter User Name"
          type="text"
          value=""
        />
      </label>
       
      <div>
        <button
          type="button"
        >
          Display My Name
        </button>
        <p />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
